src/guards/roles.guard.ts

```diff
@@ -0,0 +1,33 @@
+import {
+    Injectable,
+    CanActivate,
+    ExecutionContext,
+    Inject,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { UserService } from '../user/user.service';
+import { User } from 'src/model/user.model';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+    constructor(
+        @Inject('UserService') private readonly userService: UserService,
+        private readonly reflector: Reflector,
+    ) {}
+
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+        const roles = this.reflector.get<string[]>(
+            'roles',
+            context.getHandler(),
+        );
+        console.log('roles guard canactivate activatd');
+        console.log(roles);
```

Comment:
Remove `console.log`.

Reply:
Actually , I'm was going to remove that before things get done.